### ISSUE TITLE
Contour lines in surface plots.

### ISSUE THREAD

**Issue by MiltonCobo** [ecomfe/echarts-gl#380]
It will be very useful, when plotting 3D surfaces, to allow the projection of contour lines in the grid planes, similar to what plotly.js does. Also plotly.js has a feature Contour that plots the 2D contour lines of a surface, and I don’t know if this can be obtained with heatMap.

**Comment by MiltonCobo** [comment]
This is an example of  a surface plot with level curves on the canonical planes in plotly.js. 

Plotly.js library is very good but is heavy to load.

![image](https://user-images.githubusercontent.com/44674476/104325709-ef452800-54b6-11eb-80c4-06fcbfcec646.png)

And here an example of Contour plot of the same surface for the z-values:
![image](https://user-images.githubusercontent.com/44674476/104325999-37644a80-54b7-11eb-8107-13aeacd605f7.png)

**Comment by pissang** [comment]
Thanks for the detailed suggestion! Seems the contour is on the plane instead of 3D space. Correct me if I'm wrong

**Comment by MiltonCobo** [comment]
Yes, the contour lines are plotted in 2D. They could be projections of the
wireframe on the 3D-grid planes (as in figure1), or they could be plotted
as a scatter 2D plot for a given surface and axis (as in figure2 for the
z-axis).

I am a math teacher and contour lines are very useful for my graphs. In
fact, your library has so many features for mathematicians!
In particular few graph libraries have 3D plot features like yours.

I think maybe you can figure out a way to introduce these features without
too much turmoil in your code.

Thanks!

Milton.
